src/ui/map.js

```diff
@@ -2987,6 +2989,7 @@ class Map extends Camera {
                 if (willIdle) {
                     this.fire(new Event('idle'));
                     this._isInitialLoad = false;
+                    if (this._isInitialGeoJSONData) this._isInitialGeoJSONData = false;
```

Comment:
I think we can make the point at which we flip this flag to `false` a bit more specific.
 `willIdle` checks for `style.loaded()` and `map.isMoving()`. `style.loaded()` looks at all `SourceCaches` to check if they're loaded . (https://github.com/mapbox/mapbox-gl-js/blob/941ba0fdb8d1b945dcd7040c129bff559ed5a5e5/src/style/style.js#L416)
 
But for this flag, we don't care about map movement, we mostly just care about whether the geojson `SourceCache`'s have loaded or not.